checkUrlExists/lambda/index.js

```diff
@@ -50,10 +40,50 @@ async function checkUrl(url) {
 }
 
 async function run(url) {
-  const curlOpts = {
-    caInfo: certFilePath,
-  };
-  return curly.head(url, curlOpts);
+  const parsedUrl = Url.parse(url);
+  const protocol = parsedUrl.protocol; // Url.parse() lower cases the protocol
+  if (FTP_PROTOCOL === protocol || SECURE_FTP_PROTOCOL === protocol) {
+    const secure = SECURE_FTP_PROTOCOL === protocol;
+    const ftpClient = new ftp.Client();
+    try {
+      let options = {
+        host: parsedUrl.host,
+        secure: secure,
+      };
+      if (parsedUrl.port) {
+        options = { port: parsedUrl.port, ...options };
+      }
+      await ftpClient.access(options);
```

Comment:
If this or any other `await` fails, it will throw an exception, which will be caught on line 46/36 (note to self and anybody else who isn't familiar with JS promises and `await`.